grpcgcp/gcp_balancer_test.go

```diff
@@ -165,6 +226,81 @@ func TestCreatesMinSubConns(t *testing.T) {
 	}
 }
 
+func TestCreatesUpToMaxSubConns(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	mockCC := mocks.NewMockClientConn(mockCtrl)
+	newSCs := []*mocks.MockSubConn{}
+	mockCC.EXPECT().UpdateState(gomock.Any()).AnyTimes()
+	mockCC.EXPECT().NewSubConn(gomock.Any(), gomock.Any()).DoAndReturn(func(_, _ interface{}) (*mocks.MockSubConn, error) {
+		newSC := mocks.NewMockSubConn(mockCtrl)
+		newSC.EXPECT().Connect().AnyTimes()
+		newSC.EXPECT().UpdateAddresses(gomock.Any()).AnyTimes()
+		newSCs = append(newSCs, newSC)
+		return newSC, nil
+	}).Times(5)
+
+	b := newBuilder().Build(mockCC, balancer.BuildOptions{}).(*gcpBalancer)
+	// Simulate ClientConn calls UpdateClientConnState with the config provided to Dial.
+	b.UpdateClientConnState(balancer.ClientConnState{
+		ResolverState: resolver.State{},
+		BalancerConfig: &GcpBalancerConfig{
+			ApiConfig: &pb.ApiConfig{
+				ChannelPool: &pb.ChannelPoolConfig{
+					MinSize:                          3,
+					MaxSize:                          5,
+					MaxConcurrentStreamsLowWatermark: 1,
+				},
+			},
+		},
+	})
+
+	if want := 3; len(b.scRefs) != want {
```

Comment:
try avoid using such magic numbers and define them as const, with necessary additive values.

Reply:
Yeah, I was thinking about that. Done.